fileHub/js-app/app/application.js

```diff
@@ -42,6 +43,7 @@ export class Application extends Component {
         .addRoute('index', () => {
           new FileListPage(this.rootElement, titleService, stateManager)
               .onNavigateToFolder((folderId) => router.redirect(`index/${folderId}`));
+          stateManager.dispatch(new GetCurrentUser());
           stateManager.dispatch(new GetRootFolder());
```

Comment:
GetCurrentUser and GetRootFolder should be called only if there's no such fields in state.